nifi-extension-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/kinesis/stream/record/converter/RecordConverterWrapper.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.aws.kinesis.stream.record.converter;
+
+import org.apache.nifi.serialization.SimpleRecordSchema;
+import org.apache.nifi.serialization.record.MapRecord;
+import org.apache.nifi.serialization.record.Record;
+import org.apache.nifi.serialization.record.RecordField;
+import org.apache.nifi.serialization.record.RecordFieldType;
+import org.apache.nifi.serialization.record.RecordSchema;
+import software.amazon.kinesis.retrieval.KinesisClientRecord;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class RecordConverterWrapper implements RecordConverter {
+
+    private static final String VALUE = "value";
+    private static final String METADATA = "metadata";
+
+    private static final String STREAM = "aws.kinesis.stream";
+    private static final String SHARD_ID = "aws.kinesis.shard.id";
+    private static final String SEQUENCE_NUMBER = "aws.kinesis.sequence.number";
+    private static final String PARTITION_KEY = "aws.kinesis.partition.key";
+    private static final String APPROX_ARRIVAL_TIMESTAMP = "aws.kinesis.approximate.arrival.timestamp";
```

Comment:
On further consideration and comparison to the Kafka metadata wrapper, recommend simplifying these property names and removing the use of the `.` separator. The record is already implied to be from Kinesis, so the `aws.kinesis` prefix is not needed, and avoiding the use of the `.` separator should avoid some potential issues with downstream components parsing JSON.
```suggestion
    private static final String STREAM = "stream";
    private static final String SHARD_ID = "sharedId";
    private static final String SEQUENCE_NUMBER = "sequenceNumber";
    private static final String PARTITION_KEY = "partitionKey";
    private static final String APPROX_ARRIVAL = approximateArrival";
```

Reply:
There's a typo in SHARD_ID. Anyways, fixed